source/PlayerInfo.cpp

```diff
@@ -841,14 +841,32 @@ void PlayerInfo::AddShip(const shared_ptr<Ship> &ship)
 
 
 
-// Buy a ship of the given model, and give it the given name.
-void PlayerInfo::BuyShip(const Ship *model, const string &name)
+// Charge the player the cost of a ship and give it the specified name:
+void PlayerInfo::BuyShip(const Ship *model, const std::string &name)
+{
+	ReceiveShip(model, name, false);
+}
+
+
+
+// Receive a ship as a gift
+void PlayerInfo::GiftShip(const Ship *model, const std::string &name)
+{
+	ReceiveShip(model, name, true);
+}
+
+
+
+// Purchase or receive a gift of a ship of the given model and name
+void PlayerInfo::ReceiveShip(const Ship *model, const string &name, bool gifting)
 {
 	if(!model)
 		return;
 	
 	int day = date.DaysSinceEpoch();
-	int64_t cost = stockDepreciation.Value(*model, day);
+	int64_t cost = 0;
+	if(!gifting)
+		cost = stockDepreciation.Value(*model, day);
```

Comment:
Just one more quick suggestion.
```suggestion
	int64_t cost = gifting ? 0 : stockDepreciation.Value(*model, day);
```